src/XRP/xrp-transaction.ts

```diff
@@ -0,0 +1,110 @@
+import { Transaction } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import RippledFlags from '../rippled-flags'
+import XRPSigner from './xrp-signer'
+import XRPTransactionType from './xrp-transaction-type'
+import XRPPayment from './xrp-payment'
+import XRPMemo from './xrp-memo'
+
+/*
+ * A transaction on the XRP Ledger.
+ *
+ * @see: https://xrpl.org/transaction-formats.html
+ */
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPTransaction {
+  public static from(transaction: Transaction): XRPTransaction | undefined {
+    const account = transaction
+      .getAccount()
+      ?.getValue()
+      ?.getAddress()
+
+    const fee = transaction.getFee()?.getDrops()
+
+    const sequence = transaction.getSequence()?.getValue()
+
+    const signingPublicKey = transaction.getSigningPublicKey()?.getValue_asU8()
+
+    const transactionSignature = transaction
+      .getTransactionSignature()
+      ?.getValue_asU8()
+
+    const accountTransactionID = transaction
+      .getAccountTransactionId()
+      ?.getValue_asU8()
+
+    const flags = transaction.getFlags()?.getValue()
+
+    const lastLedgerSequence = transaction.getLastLedgerSequence()?.getValue()
+
+    let memos
```

Comment:
can assign to `undefined` here instead of having the else. Up to you

Reply:
nice, shortened